Yoast/wordpress-seo
# 537 refactor microcaching the api endpoints into using cache decorators

## Context


* We want to abstract the caching part away from getting the actual data.

## Summary


This PR can be summarized in the following changelog entry:

* Refactors microcaching to use the decorator pattern.

## Relevant technical choices:

*

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Use both the test instructions from https://github.com/Yoast/wordpress-seo/pull/21847 and https://github.com/Yoast/wordpress-seo/pull/21861 to validate this PR did not introduce any regressions.

#### Relevant test scenarios


### Test instructions for QA when the code is in the RC


* [] QA should use the same steps as above.


QA can test this PR by following these steps:

* For QA just testing the entire feature will also make sure this PR did not do anything strange.

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


## Innovation


Fixes #
